pkg/noderesourcetopology/pluginhelpers.go

```diff
@@ -0,0 +1,175 @@
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+)
+
+// topologyListerInstance is a Singleton object
+// Should not be accessed directly.
+// Should be accessed via getNodeTopologyInformer function
+var (
+	topologyListerInstance *listerv1alpha1.NodeResourceTopologyLister
+	once                   sync.Once
+)
+
+func findNodeTopology(nodeName string, nodeResTopoPlugin *nodeResTopologyPlugin) *topologyv1alpha1.NodeResourceTopology {
+	klog.V(5).Infof("namespaces: %s", nodeResTopoPlugin.namespaces)
+	for _, namespace := range nodeResTopoPlugin.namespaces {
+		klog.V(5).Infof("data.lister: %v", nodeResTopoPlugin.lister)
+		// NodeTopology couldn't be placed in several namespaces simultaneously
+		lister := nodeResTopoPlugin.lister
+		nodeTopology, err := (*lister).NodeResourceTopologies(namespace).Get(nodeName)
+		if err != nil {
+			klog.V(5).Infof("Cannot get NodeTopologies from NodeResourceTopologyNamespaceLister: %v", err)
+			continue
+		}
+		if nodeTopology != nil {
+			return nodeTopology
+		}
+	}
+	return nil
+}
+
+func initNodeTopologyInformer(masterOverride, kubeConfigPath *string) (*listerv1alpha1.NodeResourceTopologyLister, error) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(*masterOverride, *kubeConfigPath)
+	if err != nil {
+		klog.Errorf("Cannot create kubeconfig based on: %s, %s, %v", *masterOverride, *kubeConfigPath, err)
+		return nil, err
+	}
+
+	topoClient, err := topoclientset.NewForConfig(kubeConfig)
+	if err != nil {
+		klog.Errorf("Cannot create clientset for NodeTopologyResource: %s, %s", kubeConfig, err)
+		return nil, err
+	}
+
+	topologyInformerFactory := topologyinformers.NewSharedInformerFactory(topoClient, 0)
+	nodeTopologyInformer := topologyInformerFactory.Topology().V1alpha1().NodeResourceTopologies()
+	nodeResourceTopologyLister := nodeTopologyInformer.Lister()
+
+	klog.V(5).Infof("start nodeTopologyInformer")
+	ctx := context.Background()
+	topologyInformerFactory.Start(ctx.Done())
+	topologyInformerFactory.WaitForCacheSync(ctx.Done())
+
+	return &nodeResourceTopologyLister, nil
+}
+
+// getNodeTopologyLister will init v1alpha1.NodeResourceTopologyInformer once and return it.
+// if v1alpha1.NodeResourceTopologyInformer already initialized, the same instance will be return
+func getNodeTopologyLister(masterOverride, kubeConfigPath *string) (*listerv1alpha1.NodeResourceTopologyLister, error) {
```

Comment:
This is only called by New(). We don't need to introduce once() and a global variable to achieve this, do we?